sm/download/src/download.rs

```diff
@@ -72,15 +91,19 @@ impl Downloader {
         })
         .await?;
 
-        let content = response.bytes().await?;
+        let len: u64 = match response.content_length() {
+            Some(package_size) => package_size,
+            None => self.reserved_buffer_size as u64,
+        };
 
-        // Cleanup after `disc full` will happen inside atomic write function.
-        tedge_utils::fs::atomically_write_file_async(
-            self.download_target.as_path(),
-            self.target_filename.as_path(),
-            content.as_ref(),
-        )
-        .await?;
+        let mut file = File::create(self.target_filename.as_path())?;
+        // Reserve diskspace
+        let _ = fallocate(file.as_raw_fd(), FallocateFlags::empty(), 0, len as i64);
```

Comment:
To be removed.

1) This will pre allocate 5% of the disk when the server doesn't provide a content length. Even for a tiny download.
2) C8y doesn't provide a content-length.
3) This linux and even file system specific. But this is a detail compared to pint 1).